code-challenges.test.js

```diff
@@ -10,60 +18,153 @@
+        return newWords.join(" ")
+}
+console.log(wordsCapitalized(randomNouns1, randomNouns2))
 
+//I was unable to get the commas in my final output. I tried using .string(" ") to put the commas back in but no matter where I put it in the function, it was not returning the final output with commas seperating the values. 
 
 
 // --------------------3) Create a function that takes in a string and logs the index of the first vowel.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const vowelTester1 = "learn"
-// Expected output: 1
-const vowelTester2 = "academy"
-// Expected output: 0
-const vowelTester3 = "challenges"
-// Expected output: 2
-
-
-// b) Create the function that makes the test pass.
+// const vowelTester1 = "learn"
+// // Expected output: 1
+// const vowelTester2 = "academy"
+// // Expected output: 0
+// const vowelTester3 = "challenges"
+// // Expected output: 2
+
+describe("firstVowel", () => {
+    it("takes in a string and logs the index of the first vowel", () => {
+    const vowelTester1 = "learn"
+    const vowelTester2 = "academy"
+    const vowelTester3 = "challenges"
+     expect(firstVowel(vowelTester1)).toEqual(1)
+     expect(firstVowel(vowelTester2)).toEqual(0)
+     expect(firstVowel(vowelTester3)).toEqual(2)
+   })
+ })
+ //ReferenceError: firstVowel is not defined
+
+ // b) Create the function that makes the test pass.
+ 
+ //use the .indexOf method to return the index of the first instance of the given value. The given value would be the first vowel instance of each string. The strings would need to be put in an array first by using the .join() method or concatinating them into one array and defined under a different variable. Now each word has to be seperated into its own value in the array using .split(" "). 
+    //This is where I ran into the issue of not getting the words to seperate. So I decided to write a for loop for each of the variables.
+//If i write a for loop for each variable, define the vowels in a new variable and utilize .indexOf, then the index of the first vowel should be returned. 
+ //The vowels would have to be defined in a separate variable.
+ //Parameters:"learn", "academy", "challenges"
+ //Return: 1, 0, 2
+    const vowelTester1 = "learn"
+    const vowelTester2 = "academy"
+    const vowelTester3 = "challenges"
+console.log(vowelTester1.concat(vowelTester2))
+
+
+  const firstVowel = [];
+  for(let i=0, i <string.length, i++){
+    let vowels = firstVowel[] = 'aeiou';
+    console.log(vowels);
+  }
+console.log()
+//This was where I left off. I do not understand how to put into logic what I want to do using what we learned. I've tried concat, join, split, even console logging every single vowel for each variable, which worked but there is an easier way. I do not know how to do it. I know what I want to do, I just don't know how to write it. I feel like I'm using the right tools but it's piecing them together that is bringing me to literal tears. 
```

Comment:
This problem is really tricky and you were definitely heading in the right direction!  You would need some type of iteration and there are many ways to go about this problem.  You mentioned logging every vowel, which is an approach I would definitely start with, but you will also need to compare the current value with each vowel (think conditional when you know you will need to compare).  Don't get discouraged, keep working hard, doing research and sharing the excellent pseudocode!